### PR TITLE
Non-conftest nodes involved in configure checks now get node info cleared after check.

### LINKED ISSUE #2757: SConf marks files up-to-date while using them in checks
This issue was originally created at: 2011-04-23 13:50:13.
This issue was reported by: `garyo`.
garyo said at 2011-04-23 13:50:13
>If you have a `Configure` (aka `SConf`) test that uses a header file, say `foo.h`, and `foo.h` is modified, what happens is the Configure test runs (it's like a mini-SCons run) and updates the `ninfo.csig` of `foo.h`'s node.  Then the rest of the SConstruct gets read, and `foo.h` appears unmodified, so none of its dependents get built.  Don't have a repro case handy here, will try to provide one soon.

### PR DESCRIPTION
**Pull request by dmoody256**
Fixes #2757 

from https://pairlist4.pair.net/pipermail/scons-users/2022-May/008989.html

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: test/sconsign/script/Configure.py [hunk 1]
```diff
@@ -84,10 +84,10 @@
         %(sig_re)s \[.*\]
 conftest_%(sig_re)s_0_%(sig_re)s%(_obj)s:
         %(_sconf_temp_conftest_0_c)s: %(sig_re)s \d+ \d+
-        %(CC)s: %(sig_re)s \d+ \d+
+        %(CC)s: %(sig_re)s None None
```
Comment (dmoody256) [inline test/sconsign/script/Configure.py:89]: this is the compiler which is considered a non-conftest node, so if only configure checks are run, then the node info is not stored.

File: test/Configure/conftest_source_file.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+test.dir_fixture("conftest_source_file")
+
+test.run(arguments='.')
+
+test.write('header2.h', """
+#pragma once
+int test_header = 2;
+""")
+
+test.not_up_to_date(read_str="Checking for C header file header1.h... (cached) yes\n")
```
Comment (dmoody256) [inline test/Configure/conftest_source_file.py]: I tested before and after and verified this test fails with master, and passed with the PR

**Comment by dmoody256** [comment]
Not sure about the sider issue, should the code style remain consistent?

**Comment by bdbaddog** [comment]
What happens when you run scons twice in a row where a real file is used by configure, but nothing depends on it?
Does the configure step get rerun?

**Comment by bdbaddog** [comment]
> Not sure about the sider issue, should the code style remain consistent?

punt on this one. There'd be too much to fix.. We'll do a black run on the whole codebase at some point and then enforce.

**Comment by dmoody256** [comment]
> What happens when you run scons twice in a row where a real file is used by configure, but nothing depends on it?
Does the configure step get rerun?

No it uses the cached value. I updated the test to show this.

**Comment by bdbaddog** [comment]
@dmoody256 - so from looking at the code, I think the only possible functional change is that if you were to only run the configure step and no actual build steps the sconsign would be missing the content signature information for all non-configure generated files uses while processing the configure checks?

Does that sound correct?

**Comment by dmoody256** [comment]
> @dmoody256 - so from looking at the code, I think the only possible functional change is that if you were to only run the configure step and no actual build steps the sconsign would be missing the content signature information for all non-configure generated files uses while processing the configure checks?
> 
> Does that sound correct?

Yes

**Pull request merged**